src/DistributedComputations/distributed_fields.jl

```diff
@@ -197,3 +202,50 @@ for (reduction, all_reduce_op) in zip((:sum, :maximum, :minimum, :all, :any, :pr
         end
     end
 end
+
+# Distributed norm
+@inline function norm(u::DistributedField)
+    n² = dot(u, u)
+    return sqrt(n²)
+end
+
+# Distributed dot product
+@inline function dot(u::DistributedField, v::DistributedField)
+    dot_local = sum(interior(u) .* interior(v))
```

Comment:
This will probably not work because it does not remove immersed cells from the computation. We need to do something like
```suggestion
     ca = condition_operand(a, condition, 0) 
     cb = condition_operand(b, condition, 0) 
      
     B = ca * cb # Binary operation 
     r = zeros(a.grid, 1) 
      
     Base.mapreducedim!(identity, +, r, B) 
     dot_local = CUDA.@allowscalar r[1] 
```